src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -56,11 +56,52 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
         {
             // TODO: Dispatch from the caller
             await Task.Yield();
-            Exception exception = null;
+
             try
             {
                 await _lifetimeManager.OnConnectedAsync(connection);
+                // RunHubAsync does not throw but can only run if OnConnectedAsync completed successfully
+                await RunHubAsync(connection);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(0, ex, "Error when invoking OnConnectedAsync on lifetime manager.");
+            }
 
+            try
+            {
+                await _lifetimeManager.OnDisconnectedAsync(connection);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(0, ex, "Error when invoking OnDisconnectedAsync on lifetime manager.");
+            }
+        }
+
+        private async Task RunHubAsync(Connection connection)
+        {
+            Exception exception = null;
+            await HubOnConnectedAsync(connection);
+
+            try
+            {
+                await DispatchMessagesAsync(connection);
+            }
+            catch(Exception ex)
+            {
+                _logger.LogError(0, ex, "Error when processing requests.");
+                connection.Channel.Input.Complete(ex);
```

Comment:
I think we need to let errors bubble out to the middleware and handle closing the connection here